/**
 * FILE: Core.proto
 * BRIEF: Main protocol file for the SOAR Avionics System, any shared enums or messages should be defined here.
 * AUTHOR: Christopher Chan (cjchanx)
 */
syntax = "proto3";
package Proto;

/* Universal Enums ------------------------------------------------------------------*/

/* These are the nodes in the SOAR Avionics System.
 * The values are used to identify the source or destination of a message.
*/
enum Node {
    NODE_INVALID = 0;
    
    NODE_UNKNOWN = 1;   // This can happen if the node is not yet known at this stage, eg. COBS decode failed
    NODE_ANY = 2;
    NODE_RCU = 3;
    NODE_DMB = 4;
    NODE_PBB = 5;
    NODE_SOB = 6;
}

enum MessageID {
    MSG_INVALID = 0;

    MSG_UNKNOWN = 1;    // This can happen if the message ID is not yet known at this stage, eg. COBS decode failed
    MSG_CONTROL = 2;
    MSG_COMMAND = 3;
    MSG_TELEMETRY = 4;

    MSG_MAX_INVALID = 5;
}

/* Board Specific Enums ------------------------------------------------------------------*/

enum RocketState {
    DMB_INVALID = 0;

    // Current DMB RocketState (these should be kept close to the actual RocketState enum in RocketSM.hpp)
    RS_PRELAUNCH = 1;
    RS_FILL = 2;
    RS_ARM = 3;
    RS_IGNITION = 4;
    RS_LAUNCH = 5;
    RS_BURN = 6;
    RS_COAST = 7;
    RS_DESCENT = 8;
    RS_RECOVERY = 9;
    RS_ABORT = 10;
    RS_TEST = 11;
    RS_NONE = 12;
}